src/server/pfs/server/driver.go

```diff
@@ -1778,6 +1808,37 @@ nextSubvBI:
 			}
 		}
 
+		// this ensures that the job's output commit uses the latest commit on the branch, by ensuring it is the
+		// last commit to appear in the provenance slice
+		var err error
+		sort.SliceStable(newCommitInfo.Provenance, func(i, j int) bool {
+			// to make sure the parent relationship is respected during sort, we need to make sure that we organize
+			// the provenance by repo name and branch name
+			if newCommitInfo.Provenance[i].Commit.Repo.Name != newCommitInfo.Provenance[j].Commit.Repo.Name {
+				return newCommitInfo.Provenance[i].Commit.Repo.Name < newCommitInfo.Provenance[j].Commit.Repo.Name
+			} else if newCommitInfo.Provenance[i].Branch.Name != newCommitInfo.Provenance[j].Branch.Name {
+				return newCommitInfo.Provenance[i].Branch.Name < newCommitInfo.Provenance[j].Branch.Name
+			}
+
+			// we need to check the commit info of the 'j' provenance commit to get the parent
+			var provCommitInfo *pfs.CommitInfo
+			provCommitInfo, err = d.resolveCommit(stm, newCommitInfo.Provenance[j].Commit)
```

Comment:
This can still mask an error because it can set `err` on one call and then overwrite it with `nil` on another call. You should give the error a slightly longer name and assign to it explicitly only when the thing you're assigning is non-nil.